x-pack/plugins/security_solution/public/common/store/types.ts

```diff
@@ -40,7 +40,7 @@ export type StoreState = HostsPluginState &
  * `combineReducers` returns a type wrapped in `CombinedState`.
  * `CombinedState` is required for redux to know what keys to make optional when preloaded state into a store.
  */
-export type State = CombinedState<StoreState>;
+export type State = StoreState;
```

Comment:
Was this necessary? Also if we're going with this change then the comment block above needs an update

Reply:
It is necessary, yes. Using the internal CombinedState type leads to errors like this with the new version of Redux:

`Exported variable 'selectNotesByIdSelector' has or is using name '$CombinedState' from external module "[..]/kibana/node_modules/redux/index" but cannot be named.`

This solution seemed neat and works with the type returned from combineReducers. If you have a better suggestion I would be happy to implement that instead.


I removed the above comment and unified the StoreState and State types in the latest commit.

Reply:
Thanks for the explanation. This solution is indeed quite neat. 👍